### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by nathanaw** [nathanaw/muf#2]
(Submitted by jrvdboom on Codeplex)

Despite the UndoRoot containing a WeakReference to its root object, memory is still leaked by the _Roots dictionary of the UndoService, because its keys are the same root objects but then strong-referenced. Adding a RemoveRoot method or changing the dictionary to use WeakReference keys can solve the problem.

**Comment by TruePluto** [comment]
hi, nice to see this.
I'm not sure that using the **ConditionalWeakTable** replace the dictonary will be OK
```
        //private IDictionary<object, UndoRoot> _Roots;
       ConditionalWeakTable <object, UndoRoot> _Roots;
        public UndoService()
        {
             // _Roots = new Dictionary<object, UndoRoot>();
                _Roots = new ConditionalWeakTable<object, UndoRoot>();
        }
```
But I will using this pacakge, It's wonderful. 
Hope it will upgrade soon. I wish the empty changeset issue could be solved too.

**Comment by nathanaw** [comment]
This should be fixed in PR #8 and the 2.0 release.

I've converted the UndoService to use weak references for the keys.
